### PR TITLE
limereport: add library

### PR DESCRIPTION
**Pull request by Jihadist**
Specify library name and version:  **limereport/1.7.4**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Jihadist** [comment]
Blocked by #19639

**Comment by Jihadist** [comment]
Package requires qt options disabled by default in cci recipe. I have no ideas how to do with it.

File: recipes/limereport/all/conanfile.py
Comment (RubenRBS) [inline recipes/limereport/all/conanfile.py:71]:
```suggestion
        qt_major = Version(self.dependencies["qt"].ref.version).major
```

File: recipes/limereport/all/conanfile.py
Comment (RubenRBS) [inline recipes/limereport/all/conanfile.py:94]:
```suggestion
        qt_major = Version(self.dependencies["qt"].ref.version).major
```

File: recipes/limereport/all/conanfile.py
Comment (RubenRBS) [inline recipes/limereport/all/conanfile.py:48]: Can you pin libpng to a specific version? We still haven't decided on wheter to allow libpng as a version range

File: recipes/limereport/all/conanfile.py
Comment (RubenRBS) [inline recipes/limereport/all/conanfile.py:50]:
```suggestion
            self.tool_requires("zint/2.10.0")
```

File: recipes/limereport/all/conanfile.py
Comment (RubenRBS) [inline recipes/limereport/all/conanfile.py:42]: Not used?

File: recipes/limereport/all/conanfile.py
Comment (RubenRBS) [inline recipes/limereport/all/conanfile.py:20]: Missing `package_type = "library"`

File: recipes/limereport/all/conanfile.py
Comment (RubenRBS) [inline recipes/limereport/all/conanfile.py:56]: Is this C++ 11 or C++ 17?

If the lib needs something greater than 11, CCI policy is to add a check in the validate method

As per https://github.com/fralx/LimeReport/blob/master/CMakeLists.txt#L50, you should be validating the cppstd version, but maybe Im missing something!



(see the template here https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/cmake_package/all/conanfile.py#L82 for how to do so in case it's necessary)

File: recipes/limereport/all/conanfile.py
Comment (RubenRBS) [inline recipes/limereport/all/conanfile.py:45]: The project seems to be a cmake lib. Does `cmake_layout` not fit better here?

**Review by RubenRBS — COMMENTED** [review]
Thanks a lot for your contribution. Some comments :)

File: recipes/limereport/all/conanfile.py
Comment (Jihadist) [reply recipes/limereport/all/conanfile.py]: Done

File: recipes/limereport/all/conanfile.py
Comment (Jihadist) [reply recipes/limereport/all/conanfile.py:70]: Done

File: recipes/limereport/all/conanfile.py
Comment (Jihadist) [reply recipes/limereport/all/conanfile.py:22]: Done

File: recipes/limereport/all/conanfile.py
Comment (Jihadist) [reply recipes/limereport/all/conanfile.py]: I pinned. But looks freetype brought problem here ```Version conflict: Conflict between libpng/1.6.40 and libpng/1.6.42 in the graph.```

**Comment by Jihadist** [comment]
Blocked by #22683

File: recipes/limereport/all/conanfile.py
Comment (RubenRBS) [inline recipes/limereport/all/conanfile.py:95]: Why have the option if it's going to unconditionally be removed? Are you missing a `if self.options.shared:`?

File: recipes/limereport/all/test_package/test_package.cpp
Comment (RubenRBS) [inline recipes/limereport/all/test_package/test_package.cpp:2]: I'd like to know how these are ending up being found. Do they directly come from limereport? I see no requirement to QT in the test package nor are the headers transitive in the main recipe, so wondering about it

File: recipes/limereport/all/test_package/CMakeLists.txt
Comment (RubenRBS) [inline recipes/limereport/all/test_package/CMakeLists.txt:13]: These libs too, do they directly come from limereport?

File: recipes/limereport/all/test_package/CMakeLists.txt
Comment (RubenRBS) [inline recipes/limereport/all/test_package/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.15)
```

[Compatibility with CMake < 3.5 will be removed from a future version of CMake](https://cmake.org/cmake/help/latest/release/3.27.html#deprecated-and-removed-features), let's try to stick to 3.15 for tests :)

**Review by RubenRBS — COMMENTED** [review]
Thanks for your patience, some questions before being able to merge, otherwise looks good :)

File: recipes/limereport/all/test_package/test_package.cpp
Comment (Jihadist) [reply recipes/limereport/all/test_package/test_package.cpp:2]: I missed ```find_package``` for qt from cmakelists but it works locally

File: recipes/limereport/all/conanfile.py
Comment (Jihadist) [reply recipes/limereport/all/conanfile.py]: Yep, done

File: recipes/limereport/all/test_package/CMakeLists.txt
Comment (Jihadist) [reply recipes/limereport/all/test_package/CMakeLists.txt:21]: done too

File: recipes/limereport/all/test_package/CMakeLists.txt
Comment (Jihadist) [reply recipes/limereport/all/test_package/CMakeLists.txt]: done

**Comment by Jihadist** [comment]
Blocked be #22788

**Review by RubenRBS — DISMISSED** [review]

File: recipes/limereport/all/test_package/test_package.cpp
Comment (Jihadist) [inline recipes/limereport/all/test_package/test_package.cpp]: Are you sure it needed to be removed? Because LimeReport engine cannot be created without QGuiApplication

File: recipes/limereport/all/conanfile.py
Comment (Jihadist) [reply recipes/limereport/all/conanfile.py:77]: What about this https://github.com/fralx/LimeReport/blob/master/CMakeLists.txt#L384 ?

File: recipes/limereport/all/test_package/test_package.cpp
Comment (uilianries) [reply recipes/limereport/all/test_package/test_package.cpp]: @Jihadist We dont need really create an application, only check the package consistence. If we can build a very small application that proves we are able to build and run, that's enough.

File: recipes/limereport/all/conanfile.py
Comment (Jihadist) [reply recipes/limereport/all/conanfile.py:77]: Ok, maybe recipe has been changed commit from my initial commit.

File: recipes/limereport/all/test_package/test_package.cpp
Comment (Jihadist) [reply recipes/limereport/all/test_package/test_package.cpp]: Ok. I did this because old recipes contain runnable test_packages.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by perseoGI — APPROVED** [review]
LGTM! Thank for your contribution

File: recipes/limereport/all/conanfile.py
Comment (uilianries) [inline recipes/limereport/all/conanfile.py:62]:
```suggestion
        self.requires("qt/[>=6.0 <7]", transitive_headers=True, transitive_libs=True)
```

We are actually enabling version ranges for Qt. It should avoid not only conflicts, but improve the build time when packaging new versions and dropping older versions.

Comment (uilianries) [reply recipes/limereport/all/conanfile.py]: @maksim-petukhov Qt5 can be applied by using [replace_requires](https://docs.conan.io/2/reference/config_files/profiles.html#replace-requires) :)

File: recipes/limereport/all/conanfile.py
Comment (uilianries) [inline recipes/limereport/all/conanfile.py:62]:
```suggestion
        self.requires("qt/[>=5.15 <7]", transitive_headers=True, transitive_libs=True)
```

File: recipes/limereport/all/conanfile.py
Comment (uilianries) [reply recipes/limereport/all/conanfile.py]: @maksim-petukhov I just updated to support both majors. Thank you!

**Review by perseoGI — APPROVED** [review]

**Comment by uilianries** [comment]
Just add a new update in the recipe: https://github.com/conan-io/conan-center-index/pull/17841/commits/2793e6f057a22875a2dbc49edabaa9e0c7a5f161

The zint project is marked as private by limereport because is built as vendorized project and as static library. Still, it's a regular requirement, not a build requirement. Also, zint has an option to generate QZint, it requires Qt as dependency, but is not enabled by default.

**Review by danimtb — APPROVED** [review]

**Review by perseoGI — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Review by RubenRBS — APPROVED** [review]
Great job @uilianries getting this PR thru to the finish line, I know it's been quite hard :)

**Pull request merged**
